packages/docusaurus/src/server/configValidation.ts

```diff
@@ -158,9 +158,7 @@ const SiteUrlSchema = Joi.string()
     try {
       const {pathname} = new URL(value);
       if (pathname !== '/') {
-        helpers.warn('docusaurus.configValidationWarning', {
-          warningMessage: `The url is not supposed to contain a sub-path like '${pathname}'. Please use the baseUrl field for sub-paths.`,
-        });
+        return helpers.error('docusaurus.configValidationWarning');
```

Comment:
I don't think we should use `configValidationWarning` for something that is an error. Also `pathname` used to be interpolated but it's not anymore

Unfortunately I can't really guide you to the correct solution: Joi is really a pain to do anything basic such as returning a custom error message 😅